.github/workflows/main.yaml

```diff
@@ -3,6 +3,9 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    paths:
+      - '.github/workflows/cpp.yml'
```

Comment:
Oh neat! So, to confirm my understanding... this runs `main.yaml` anytime the file is changed on a pull request?

Reply:
Correct--you can see that it just ran now that I fixed it. So it still wouldn't have caught the change in #60 since I didn't touch this file. But if we ever edit this file in the future, the PR will run it, we won't have to wait for it to land on main. 

Since this is basically testing the Makefile, we could also have it run when the Makefile is modified?

Reply:
That makes sense. I like the idea of workflows being task-focused instead of branch-focused. e.g., `makefile.yaml` instead of `main.yaml` or something...